src/main/java/com/datapipe/jenkins/vault/model/VaultSecretValue.java

```diff
@@ -41,16 +41,25 @@
 
     private String envVar;
     private final String vaultKey;
+    private final boolean isRequired;
 
     @Deprecated
     public VaultSecretValue(String envVar, @NonNull String vaultKey) {
         this.envVar = fixEmptyAndTrim(envVar);
         this.vaultKey = fixEmptyAndTrim(vaultKey);
+        this.isRequired = true; // Old behavior
     }
 
     @DataBoundConstructor
     public VaultSecretValue(@NonNull String vaultKey) {
         this.vaultKey = fixEmptyAndTrim(vaultKey);
+        this.isRequired = false;
+    }
+
+    @DataBoundConstructor
+    public VaultSecretValue(@NonNull String vaultKey, boolean required) {
+        this.vaultKey = fixEmptyAndTrim(vaultKey);
+        this.isRequired = required;
```

Comment:
You can only have one `@DataBoundConstructor`
```suggestion
```

You need to add a `@DataBoundSetter` for `isRequired

Reply:
Sorry, shooting in the dark on most of this, but doing my best with it 😆